### ISSUE TITLE
Use sbt's insideCI to enable Scala2 optimizer/inliner only when running in CI

### ISSUE THREAD

**Issue by mdedetrich** [pjfanning/sbt-pekko-build#9]
I just learnt of this [neat new trick](https://github.com/sbt/sbt-osgi/pull/90#discussion_r1468232746), sbt has a setting called `insideCI` which correctly determines if sbt is run in a CI environment or not.

We can use this setting to only enable the optimizer/inliner when running in CI or if a cli knob/toggle is enabled. Its best to do this when start releasing artifacts from CI since it will make the release process more simple

**Comment by pjfanning** [comment]
During releases, release managers (people) will need to force inline=true.

**Comment by mdedetrich** [comment]
Yes, so the question is should we just do this now? Originally I wanted to wait to do this until we get the green light to do release in CI like log4j does so that release managers wouldn't need to pass the `inline=true` flag but considering that https://github.com/apache/incubator-pekko-http/pull/456 was created the QoL/user experience isn't ideal right now.

Alternately we can document the https://github.com/pjfanning/sbt-pekko-build/blob/main/src/main/scala/com/github/pjfanning/pekkobuild/PekkoInlinePlugin.scala#L56 option and even put it on welcome screen but the more I think about it, we should just have inline only running in CI
